libs/common/src/auth/services/token.service.ts

```diff
@@ -586,8 +601,12 @@ export class TokenService implements TokenServiceAbstraction {
           // TODO: PM-6408
           // 2024-02-20: Remove refresh token from memory and disk so that we migrate to secure storage over time.
           // Remove these 2 calls to remove the refresh token from memory and disk after 3 months.
-          await this.singleUserStateProvider.get(userId, REFRESH_TOKEN_DISK).update((_) => null);
-          await this.singleUserStateProvider.get(userId, REFRESH_TOKEN_MEMORY).update((_) => null);
+          await this.singleUserStateProvider.get(userId, REFRESH_TOKEN_DISK).update(() => null, {
+            shouldUpdate: (previousValue) => previousValue !== null,
+          });
+          await this.singleUserStateProvider.get(userId, REFRESH_TOKEN_MEMORY).update(() => null, {
+            shouldUpdate: (previousValue) => previousValue !== null,
+          });
```

Comment:
```suggestion
          await this.singleUserStateProvider.get(userId, REFRESH_TOKEN_MEMORY).update((_) => null);
```
Not my ownership, but the linked epic describing the change clearly states that this applies to disk I/O only. Subsequently, there is no reason to apply it to memory state.